pkg/arvo/app/contact-view.hoon

```diff
@@ -28,20 +34,27 @@
     ^-  (quip card _this)
     :_  this
     :~  [%pass /updates %agent [our.bowl %contact-store] %watch /updates]
-        
         (contact-poke:cc [%create /~/default])
         (group-poke:cc [%bundle /~/default])
         (contact-poke:cc [%add /~/default our.bowl *contact])
         (group-poke:cc [%add [our.bowl ~ ~] /~/default])
+        :*  %pass  /srv  %agent  [our.bol %file-server]
+            %poke  %file-server-action
+            !>([%serve-dir /'~groups' /app/landscape %.n])
+        ==
     ==
   ::
-  ++  on-save   on-save:def
+  ++  on-save   !>(state)
   ++  on-load
     |=  old=*
     ^-  (quip card _this)
-    :_  this
+    :_  this(dir-served %.y)
```

Comment:
Not sure why we need to store such a flag in state. It'd be no problem to just check for the pre-`%0` version (vase contains only `~`), and only run this if we're upgrading from that.

As it stands, the upgrade logic will get run on every `+on-load`, which may be undesirable, especially since it also `%disconnect`s some eyre endpoint that might have gotten reused elsewhere.

Reply:
I think you're fine to remove the flag from state, with your new changes, the logic only gets run once. (The `+on-init` path doesn't even set it, anyway.)